lib/OpenQA/WebAPI/Controller/API/V1/User.pm

```diff
@@ -0,0 +1,60 @@
+# Copyright (C) 2020 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+
+package OpenQA::WebAPI::Controller::API::V1::User;
+use Mojo::Base 'Mojolicious::Controller';
+
+=pod
+
+=head1 NAME
+
+OpenQA::WebAPI::Controller::API::V1::User
+
+=head1 SYNOPSIS
+
+  use OpenQA::WebAPI::Controller::API::V1::User;
+
+=head1 DESCRIPTION
+
+Implements User API.
+
+=head1 METHODS
+
+=over 4
+
+=item delete()
+
+Deletes an existing user.
+
+=back
+
+=cut
+
+sub delete {
+    my ($self) = @_;
+
+    my $users = $self->schema->resultset('Users');
+    my $user  = $users->find($self->param('id'));
+
+    return $self->render(json => {error => 'Not found'}, status => 404) unless $user;
+
+    my $nickname = $user->{nickname};
+    my $ret      = $user->delete();
+
+    $self->emit_event('user_deleted', {nickname => $nickname});
+    $self->render(json => {result => int($ret)});
```

Comment:
As you are only using the variables "users", "nickname" and "ret" once I suggest to use them in place and get rid of some arbitrary blank lines:

```perl
    my ($self) = @_;
    my $user = $self->schema->resultset('Users')->find($self->param('id'));
    return $self->render(json => {error => 'Not found'}, status => 404) unless $user;
    $self->emit_event('user_deleted', {nickname => $user->{nickname}});
    $self->render(json => {result => $user->delete()});
```